### ISSUE TITLE
Sorting dates no longer works

### ISSUE THREAD

**Issue by furkick** [dj-fiorex/angular2-smart-table#96]
The table is no longer able to sort dates. I have tried a number of different formats with no success. 

Please see this [StackBlitz example](https://stackblitz.com/edit/angular-ivy-ag3itp)

**Comment by uap-universe** [comment]
### TL;DR

If you can give me a function that *reliably* checks if an object can be interpreted as a number (i.e. either *is* a number or is a string that represents a valid number), I will use that to fix the bug. Read on if you want to know why this bug even exists.

### Rant

In the past, the table was accidentally (!) able to sort dates, because the original developers did not care for type safety and just used less than to compare whatever the could get their hands on (including null values), which was incorrect in particular when null values were present.

> The current implementation is type safe and compares values according to the following logic.
> 
> If both values can be interpreted as a number (or null), they are compared as if they were numbers.
> If both values are either null, undefined or typeof string, they are compared as strings using the current locale.
> Otherwise, they are compared using their natural ordering.
> 
> Null values are considered less than any non-null element. Null and undefined are considered equal.

Your "dates" are no dates, they are all strings, therefore they are never compared as dates.

So I would **expect** that they are compared as strings and therefore sorted lexicographically.

However, the following happens instead:

`parseFloat('2021-11-17T23:10:05Z')` is expected to be `NaN`. But it surprisingly is just `2021`. 

However, when comparing the strings, they are converted to Number by `Number(x)` and *that* correctly gives `nan`.

The documentation of `parseFloat()` says:

> The parseFloat() function parses an argument (converting it to a string first if needed) and returns a floating point number.
> Params:
> string – The value to parse. If this argument is not a string, then it is converted to one using the ToString abstract operation. Leading whitespace  in this argument is ignored.  
> Returns:
> A floating point number parsed from the given string.
> Or NaN when the first non-whitespace character cannot be converted to a number

From that documentation - at least I - would never expect that this function also returns a valid number if only the prefix of the given string is convertible to a number. What a crazy implementation....

Anyway. The fix, however, is not to use the `Number` constructor, because that gives `0` if you put in `''` (empty strings) - of course.

Just using `Number.isNaN` also does not work, because `Number.isNaN({})` is *obviously* false, because `{}` is a number, of course, namely the number 0.

`Number.isFinite()` also does not work in our case, because we actually want strings that are (reasonably) convertible to number to be interpreted and compared as numbers.

**Comment by furkick** [comment]
Not a problem, I have utilised the compareFunction for now to sort the dates myself.

Perfect explanation thank you! Appreciate it.
